fastlane/lib/fastlane/actions/hockey.rb

```diff
@@ -27,24 +27,51 @@ def self.upload(api_token, ipa, options)
         create_update = options.delete(:create_update)
 
         if create_update
-          self.create_and_update_build(api_token, ipa, options)
+          response = self.create_and_update_build(api_token, ipa, options)
         else
-          self.upload_build(api_token, ipa, options)
+          response = self.upload_build(api_token, ipa, options)
+        end
+
+        case response.status
+        when 200...300
+          url = response.body['public_url']
+
+          Actions.lane_context[SharedValues::HOCKEY_DOWNLOAD_LINK] = url
+          Actions.lane_context[SharedValues::HOCKEY_BUILD_INFORMATION] = response.body
+
+          UI.message("Public Download URL: #{url}") if url
+          UI.success('Build successfully uploaded to HockeyApp!')
+        else
+          if response.body.to_s.include?("App could not be created")
+            UI.user_error!("Hockey has an issue processing this app. Please confirm that an app in Hockey matches this IPA's bundle ID or that you are using the correct API upload token. If error persists, please provide the :public_identifier option from the HockeyApp website. More information https://github.com/fastlane/fastlane/issues/400")
+          else
+            UI.user_error!("Error when trying to upload file(s) to HockeyApp: #{response.status} - #{response.body}")
+          end
+        end
+      end
+
+      def self.upload_ipa_dsym(api_token, app_id, bundle_version, options)
+        connection = self.connection(options)
+        response = connection.put do |req|
+          req.options.timeout = options.delete(:timeout)
+          req.url("/api/2/apps/#{app_id}/app_versions/#{bundle_version}")
+          req.headers['X-HockeyAppToken'] = api_token
+          req.body = options
+        end
+
+        if response.status == 201
```

Comment:
Note that this introduces an inconsistency in the code where only this call checks for the appropriate return code. Not sure why the previous one used ranges. Is it related to the way Faraday handle redirects?

Have we tested this in both cdn vs no cdn use? See `bypass_cdn` option.

The other question is do we want to change the other response status checks?